Спринт 9

Melifaro2021 opened the pull request.

MaksBujanov committed:
Шрифты

MaksBujanov committed:
Шрифты для MACOSX

MaksBujanov committed:
Подключил шрифты. убрал капс. изменил толщину линии начертания текста кнопок. добавил цветовые схемы

MaksBujanov committed:
Настроил светлую и темную темы.

MaksBujanov committed:
Сохранил в dimens.xml размеры :
1. Отступов
2. Шрифтов
3. Иконок

MaksBujanov committed:
1. изменил названия в соответствии со snake_case
2. Убрал poster.png
3. изменил названия для Intent
4. убрал закоментированный код и др

MaksBujanov committed:
Тест 1: Пробую перекрасить Svg иконки в зависимости от темы.
activity_settings.xml
android:src="@drawable/ico_arr_back"

MaksBujanov committed:
Перекрасил svg иконки для темной темы

MaksBujanov committed:
мелочи

MaksBujanov committed:
Обновил студию до версии Android Studio Giraffe | 2022.3.1 Patch 2

MaksBujanov committed:
Начал делать страницу поиска

MaksBujanov committed:
Начал делать страницу поиска

MaksBujanov committed:
Подписал все свойства в XML разметке
начал страницу поиска, сделал кнопку назад, обновил градл и др.

MaksBujanov committed:
Сделал уроки EditText и TextWatcher

MaksBujanov committed:
Переопределил onSaveInstanceState,onRestoreInstanceState
Спринт 9/25 → Тема 4/5: Хранение данных. Начало → Урок 3/4

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/MainActivity.kt:
```diff
@@ -15,26 +13,26 @@ class MainActivity : AppCompatActivity() {
         val searchButton = findViewById<Button>(R.id.search)
         searchButton.setOnClickListener {
-            val displayIntent = Intent(
+            val goToSearch = Intent(
```
Так как это интент на нажатие кнопки Поиск, то лучше назвать searchIntent
Ну и соответственно ниже так же

Melifaro2021 replied:
Intent переводится как намерение. Название searchIntent переведем как намерение поиска. У нас кнопка делает не намерение поиска, а намерение перехода к странице Поиска поэтому goToSearch лучше отображает смысл. Рассмотрим другие Intent.  mediaIntent или settingsIntent из этих названий не ясно, что будет происходить, какое само намерение , ясно только то что оно относится к страницам media и settings. Мои названия goToMedia goToSettings отображают и к каким страницам это относится и что будет происходить после клика. Так что пока нет  других аргументов мои названия более обоснованные, оставляю их как были.

AleksandrIlinskii replied:
Тогда переименуй в navigateToSearchScreen и дальше по аналогии.

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/MediatekaActivity.kt:
```diff
@@ -2,10 +2,18 @@ package ru.makskim.playlist_maker
+
+
+
+
```
Все эти лишние строки надо удалить

Melifaro2021 replied:
Данная страница не участвует в 9 спринте. Пробелы убрал.

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SearchActivity.kt:
```diff
@@ -1,11 +1,104 @@
+
+        val linearLayout = findViewById<FrameLayout>(R.id.search_field)
+        val inputEditText = findViewById<EditText>(R.id.input_edit_text)
+        val xIco = findViewById<ImageView>(R.id.x_ico)
```
Я так понимаю, что это крестик в поле ввода.
Лучше назвать это clearSearchFieldIcon

Melifaro2021 replied:
1. Предыдущий ревьюер ставил мне в замечание , то что в проекте для переменных используется одновременно и camelCase и snake_case и предложил выбрать между двумя стандартами, я выбрал snake_case и провел работу, чтобы по всему проекту оставить один формат записи. Может быть где-то еще есть отдельные  не убранные вещи , я буду их продолжать убирать, где увижу. clearSearchFieldIcon мне не подходит из-за формата записи. Можно было представить название в виде clear_search_field_icon.
2. x_ico короче по записи, чем clear_search_field_icon
3. clear_search_field_icon вероятнее будет понятнее смысл чем x_ico для другого человека, если кто-то будет читать мой код
так что оставим clear_search_field_icon

Melifaro2021 replied:
Ok

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SearchActivity.kt:
```diff
@@ -1,11 +1,104 @@
+        super.onRestoreInstanceState(savedInstanceState)
+        restored_query = savedInstanceState.getString(KEY, DEF_SEARCH)
     }
+    /* Хранение данных END */
```
На первых порах можно писать комментарии, которые объясняют логику программы. 
Но в будущем это надо удалить

Melifaro2021 replied:
Да. сейчас я еще учусь.

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SearchActivity.kt:
```diff
@@ -1,11 +1,104 @@
+
+    // В Kotlin для создания константной переменной мы используем companion object.
+    // Ключ должен быть константным, чтобы мы точно знали, что он не изменится
+    companion object {
```
По котлин код стайлу companion object располагается в самом низу файла

Melifaro2021 replied:
Не знал, подвинул. А есть какие-то технические обоснования , зачем так делают? ![image](https://github.com/Melifaro2021/Playlist_Maker/assets/87600895/8fc963e9-6531-43a7-b26c-61d1b39ee84b)

Melifaro2021 replied:
![image](https://github.com/Melifaro2021/Playlist_Maker/assets/87600895/8fc963e9-6531-43a7-b26c-61d1b39ee84b)
![изображение_2023-11-02_132428288](https://github.com/Melifaro2021/Playlist_Maker/assets/87600895/375e9560-3ec6-43a9-a31d-7524e6e9771c)

AleksandrIlinskii replied:
Скорее вопрос стилистики структуры страницы, я был на проекте где он был сверху, сразу после конструктора.
Об этом можно почитать тут https://kotlinlang.org/docs/coding-conventions.html#class-layout

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SettingsActivity.kt:
```diff
@@ -16,5 +18,41 @@ class SettingsActivity : AppCompatActivity() {
+            val support_theme = resources.getString(R.string.theme_mail)
+            shareIntent.data = Uri.parse("mailto:")
+
+            shareIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf("melifaro00@yandex.ru"))
```
Почту тоже лучше вынести в ресурсы

Melifaro2021 replied:
Не догадался, вынес.

AleksandrIlinskii commented on app/src/main/res/layout/activity_mediateka.xml:
```diff
@@ -6,5 +6,46 @@
+            android:text="Нажми на меня!"
+            style="Widget.MaterialComponents.Button.UnelevatedButton"
+               />
+    </FrameLayout-->
```
Закомментированный код не стоит пушить

Melifaro2021 replied:
Убрал. Но экран не относится к 9 спринту и проверяться сейчас не должен. Заданий на медиатеку еще не было и я просто экспериментировал со стилями, чтобы не создавать отдельный проект для экспериментов.

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/MainActivity.kt:
Не хватает пустой строки после package на 1 строке

Melifaro2021 replied:
Поставил. Не принципиально.

AleksandrIlinskii replied:
Принципиально. Нужно разделять блоки на странице

AleksandrIlinskii commented on app/src/main/res/layout/activity_mediateka.xml:
```diff
@@ -6,5 +6,46 @@
+    </LinearLayout>
+</androidx.constraintlayout.widget.ConstraintLayout>
+
+<!-- style="@style/Widget.MaterialComponents.Button.OutlinedButton"   -->
```
На экране медиатеки ничего не должно быть

Melifaro2021 replied:
Исправил.

AleksandrIlinskii commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -6,5 +6,31 @@
     android:layout_height="match_parent"
     tools:context=".SearchActivity">
-
+    <LinearLayout style="@style/page_search" >
```
style с новой строки

AleksandrIlinskii replied:
Зачем тебе LinearLayout внутри ConstraintLayout, если ты  его не привязываешь к краям
<img width="727" alt="image" src="https://github.com/Melifaro2021/Playlist_Maker/assets/129396632/8f0e80ee-71f1-4c05-bc40-ad3eea0b98fb">

Melifaro2021 replied:
У меня еще нет четкого понимания , когда надо использовать различные конструкции и чем руководствоваться при их выборе LinearLayout или FrameLayout или contraintlayout . Если логика выбора где-то описывалась в учебнике , когда надо выбирать одно или другое, то я это пропустил... Буду рад , если укажешь , где почитать от чего зависит выбор.

Если убрать LinearLayout все элементы собьются в кучу. Как тогда убрать LinearLayout, чтобы макет сохранил свой внешний вид?
![image](https://github.com/Melifaro2021/Playlist_Maker/assets/87600895/8f0c5132-72d6-41fc-965b-2c14446980de)

AleksandrIlinskii replied:
Почитай список статей от этой и ниже, там рассказывается о каждом типе разметки

AleksandrIlinskii replied:
https://metanit.com/java/android/3.8.php

AleksandrIlinskii commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -6,5 +6,31 @@
+        <!-- Поле поиска -->
+        <LinearLayout style="@style/page_search_search_string">
+            <!-- Обертка -->
+            <FrameLayout android:id="@+id/search_field"
```
id с новой строки

Melifaro2021 replied:
Сделал.

AleksandrIlinskii commented on app/src/main/res/values-night/colors.xml:
```diff
@@ -0,0 +1,30 @@
+    <color name="blue">#2DB1ED</color>
+    <color name="blueDark">#3772E7</color>
+    <color name="gray">#737373</color>
+    <color name="pink">#E66893</color>
```
У тебя в файле ресурсов
4 раза определен FFFFFFFF
6 раз определен 1A1B22
Зачем?

Melifaro2021 replied:
Первая и вторая пара совпадают в светлой теме, но отличаются значениями в темной теме. т.е. не одно и тоже
1) values/colors.xml           <color name="txt_color">#1A1B22</color>
    values-night/colors.xml  <color name="txt_color">#1A1B22</color>
2) values/colors.xml            <color name="txt_Settings_Page">#1A1B22</color>
    values-night/colors.xml   <color name="txt_Settings_Page">#FFFFFFFF</color>
Вторая и третья пара совпали по цветам в обоих темах , но в одном случае второй пары это цвет текста, а в случае третьей пары это цвет иконок на странице. Плохая практика объединять объекты в одно свойство , даже если в текущем варианте они совпадают. Завтра дизайнеру захочется сделать иконки другим цветом, и ты заменишь цвет, а поменяют цвет не только  иконки, но и цвет текста на странице, но нам не просили его менять, это разные объекты т.е. будет ошибка из-за объединения объектов по совпадающему признаку. Это чисто мой опыт верстки на десктопе. Поэтому так.
3) values/colors.xml             <color name="ico_settings_page">#1A1B22</color>
    values-night/colors.xml   <color name="ico_settings_page">#FFFFFFFF</color>

AleksandrIlinskii commented on app/src/main/res/values/colors.xml:
```diff
@@ -1,10 +1,30 @@
     <color name="blueDark">#3772E7</color>
     <color name="gray">#737373</color>
-    <color name="pink">#E66893</color>
+    <color name="pink">#fff</color>
```
Такого цвета нет 
<img width="836" alt="image" src="https://github.com/Melifaro2021/Playlist_Maker/assets/129396632/77b036a2-13a8-4df1-8ab1-7a9fbd536287">

AleksandrIlinskii commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,15 +1,40 @@
+
+    <string name="theme_mail">
+        «Сообщение разработчикам и разработчицам приложения Playlist Maker»
+    </string>
```
Это все делается в 1 строку

Melifaro2021 replied:
Сделал.

Melifaro2021 replied:
Сделал.

AleksandrIlinskii commented:
![image](https://github.com/Melifaro2021/Playlist_Maker/assets/129396632/56c3d296-cc4c-48a3-8db4-c8b65856cc96)
Цвет текста не соответствует макетам 
<img width="179" alt="image" src="https://github.com/Melifaro2021/Playlist_Maker/assets/129396632/c392dbb1-e859-40f1-a934-2ca38021fccb">

AleksandrIlinskii commented:
Экран поиска не соответствует макету
![image](https://github.com/Melifaro2021/Playlist_Maker/assets/129396632/2fef184c-de7d-4542-bab0-7483acb0415d)

AleksandrIlinskii commented on app/src/main/java/ru/makskim/playlist_maker/SettingsActivity.kt:
```diff
@@ -16,5 +18,41 @@ class SettingsActivity : AppCompatActivity() {
             val arrToMainIntent = Intent(this, MainActivity::class.java)
             startActivity(arrToMainIntent)
         }
```
Тут нужно вызывать finish() вместо того, что сейчас есть, иначе ты создаешь стек из разных активити и это родит баги в навигации

Melifaro2021 replied:
Исправил. Да все верно. Я забыл поменять реализацию для стрелки на этом экране. На странице поиска она уже была сделана при помощи finish()

Melifaro2021 replied:
Исправил цвета текста для страницы Настроек и поиска.

MaksBujanov committed:
Исправляю замечания к Спринту 9 версия 1.

MaksBujanov committed:
Исправляю замечания к Спринту 9 версия 2.
Переписал верстку строки поиска при помощи constraintlayout. Сделал скрытие клавиатуры и  др.

AleksandrIlinskii commented:
Появился визуальный баг с полем поиска, прикладываю видео 
[Screen_recording_20231108_090713.webm](https://github.com/Melifaro2021/Playlist_Maker/assets/129396632/b122ddf0-0bfb-4a76-a1fb-480aa88641bb)
Не смотри на дерганье экрана, это прикол моего эмулятора, на моем телефоне тоже самое

MaksBujanov committed:
Исправил баг с высотой строки поиска

MaksBujanov committed:
10 Спринт.добавил:
1. SearchViewHolder.kt view песни
2. activity_song.xml разметка
3. Track.kt дата класс

MaksBujanov committed:
10 Спринт.добавил:
1. Glide.with в SearchViewHolder.kt
2. SearchAdapter.kt
3. RecyclerView
4. object MockPlaylist

MaksBujanov committed:
10 Спринт.добавил:
1. Ночную тему
2. Вынес размеры, строки
